### PR TITLE
folly: update to version 2019.06.17.00

### PR DESCRIPTION
**Pull request by beauby**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.14.5 18F132
Xcode 10.2.1 10E1001

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`? *No configured tests in Portfile*
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

File: devel/folly/Portfile [hunk 1]
```diff
@@ -16,11 +16,9 @@ maintainers         nomaintainer
 description         An open-source C++ library developed and used at Facebook.
 long_description    ${description}
 
-checksums           rmd160  8ab6f6652a14a34610b27ca97b70476400f9defe \
-                    sha256  73c5bfa832ff9e1053dd3c3010850280da111f7e75b36b6c86cc791e22a684f9 \
-                    size    2318525
-
-patchfiles          patch-cmakelists.diff patch-cmake-config.diff
```
Comment (mf2k) [inline devel/folly/Portfile]: Since this patchfile is no longer being used, it should be deleted.

Comment (beauby) [reply devel/folly/Portfile]: Done.

**Comment by beauby** [comment]
Build is failing because the old `AppleClang` compilers do not seem to implement `std::as_const` with `std=c++1z`. Any pointers to a list of compilers to black/whitelist?

File: devel/folly/Portfile (hunk 1 above)
Comment (mf2k) [reply devel/folly/Portfile]: patchfiles plural. Thank you.

**Comment by mf2k** [comment]
Sorry if I merged this too soon.

**Comment by beauby** [comment]
@mf2k All good – I initially intended to have this PR merged before #4814 but this works as well.

**Pull request merged**